Pony/include/pony/Socket.hpp

```diff
@@ -10,32 +10,41 @@
 #define PONY_SOCKET_HPP
 
 #include <pony/Address.hpp>
-#include <pony/SharedMacros.hpp>
+
+#if defined(_WIN32)
+    #include <winsock2.h>
+#endif
 
 namespace pony
 {
 
-class PONY_EXPORT Socket
-{
+bool SocketsInit();
 
+void SocketsKill();
+
+class Socket
+{
 public:
     Socket();
+    ~Socket();
 
     bool IsOpen() const;
 
-    bool Open(unsigned short port);
+    bool Open(uint16_t port);
     void Close();
 
-    int Send(const Address & dst, const void * data, unsigned size);
-    int Recv(Address & sender, void * data, unsigned size);
-
+    int32_t Send(const Address& destination, const void* data, uint32_t size);
```

Comment:
Size of Address is around 48 bits plus alignment. It's cheaper to pass by value, than by const reference.